commands/operator-sdk/cmd/generate/openapi.go

```diff
@@ -0,0 +1,173 @@
+func OpenAPIGen() error {
+	projutil.MustInProjectRoot()
+
+	absProjectPath := projutil.MustGetwd()
+	repoPkg := projutil.CheckAndGetProjectGoPkg()
+	vendor := filepath.Join(absProjectPath, "vendor")
+	srcDir := filepath.Join(vendor, "k8s.io", "kube-openapi")
+	bpFile := filepath.Join(vendor, "k8s.io", "gengo", "boilerplate", "boilerplate.go.txt")
+	binDir := filepath.Join(absProjectPath, scaffold.BuildBinDir)
+
+	if err := buildOpenAPIGenBinary(binDir, srcDir); err != nil {
+		return err
+	}
+
+	gvMap, err := genutil.ParseGroupVersions()
+	if err != nil {
+		return fmt.Errorf("failed to parse group versions: (%v)", err)
+	}
+	gvb := &strings.Builder{}
+	for g, vs := range gvMap {
+		gvb.WriteString(fmt.Sprintf("%s:%v, ", g, vs))
+	}
+
+	log.Infof("Running OpenAPI code-generation for Custom Resource group versions: [%v]\n", gvb.String())
+
+	apisPkg := filepath.Join(repoPkg, scaffold.ApisDir)
+	fqApiStr := genutil.CreateFQApis(apisPkg, gvMap)
+	fqApis := strings.Split(fqApiStr, ",")
+	if err := openAPIGen(binDir, bpFile, fqApis); err != nil {
+		return err
+	}
+
+	s := &scaffold.Scaffold{}
+	cfg := &input.Config{
+		Repo:           repoPkg,
+		AbsProjectPath: absProjectPath,
+		ProjectName:    filepath.Base(absProjectPath),
+	}
+	crdMap, err := getCRDGVKMap()
+	if err != nil {
+		return err
+	}
+	for g, vs := range gvMap {
+		for _, v := range vs {
+			gvks := crdMap[filepath.Join(g, v)]
+			for _, gvk := range gvks {
+				r, err := scaffold.NewResource(filepath.Join(gvk.Group, gvk.Version), gvk.Kind)
+				if err != nil {
+					return err
+				}
+				err = s.Execute(cfg,
+					&scaffold.CRD{Resource: r, IsOperatorGo: projutil.IsOperatorGo()},
+				)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	log.Info("Code-generation complete.")
+	return nil
+}
+
+func buildOpenAPIGenBinary(binDir, codegenSrcDir string) error {
+	genDirs := []string{"./cmd/openapi-gen"}
+	return genutil.BuildCodegenBinaries(genDirs, binDir, codegenSrcDir)
+}
+
+func openAPIGen(binDir, headerFile string, fqApis []string) error {
+	cgPath := filepath.Join(binDir, "openapi-gen")
+	for _, fqApi := range fqApis {
+		args := []string{
+			"--input-dirs", fqApi,
+			"--output-package", fqApi,
+			"--output-file-base", "zz_generated.openapi",
+			"--go-header-file", headerFile,
+		}
+		err := projutil.ExecCmd(exec.Command(cgPath, args...))
+		if err != nil {
+			return fmt.Errorf("failed to perform code-generation: %v", err)
+		}
+	}
+	return nil
+}
+
+func getCRDGVKMap() (map[string][]metav1.GroupVersionKind, error) {
+	crdInfos, err := ioutil.ReadDir(scaffold.CRDsDir)
+	if err != nil {
+		return nil, err
+	}
+	crdMap := make(map[string][]metav1.GroupVersionKind)
+	for _, info := range crdInfos {
+		if filepath.Ext(info.Name()) == ".yaml" {
+			path := filepath.Join(scaffold.CRDsDir, info.Name())
+			b, err := ioutil.ReadFile(path)
+			if err != nil {
+				return nil, err
+			}
+			crd := new(apiextv1beta1.CustomResourceDefinition)
```

Comment:
```suggestion
			crd := &apiextv1beta1.CustomResourceDefinition{}
```
As we discussed before about not using new as a convention.